src/Core/Managed/Shared/Extensibility/Implementation/TaskTimer.cs

```diff
@@ -110,26 +135,20 @@ public void Start(Func<Task> elapsed)
                     CancellationToken.None,
                     TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
                     TaskScheduler.Default);
-
-            CancelAndDispose(Interlocked.Exchange(ref this.tokenSource, newTokenSource));
         }
 
         /// <summary>
         /// Cancels the current task.
         /// </summary>
         public void Cancel()
         {
-            CancelAndDispose(Interlocked.Exchange(ref this.tokenSource, null));
+            DelayedWork currentWork = this.latestWork;
+            if (currentWork != null)
+            {
+                currentWork.IsCancellationRequested = true;
+            }
         }
 
-        /// <summary>
-        /// Releases unmanaged and - optionally - managed resources.
-        /// </summary>
-        public void Dispose()
```

Comment:
To keep functionality of classes that use `TaskTimer` "as is", they should now `Cancel` their timers instead of `Disposing` them. Simply removing the `Dispose` calls drops the `Cancel` logic of the previous implementation.